lib/segment/src/index/field_index/full_text_index/postings_iterator.rs

```diff
@@ -23,31 +21,40 @@ pub fn intersect_postings_iterator<'a>(
     Box::new(and_iter)
 }
 
-pub fn intersect_compressed_postings_iterator<'a>(
-    mut postings: Vec<ChunkReader<'a>>,
+pub fn intersect_compressed_postings_iterator<'a, H: ValueHandler + 'a>(
+    mut postings: Vec<PostingListView<'a, H>>,
     filter: impl Fn(PointOffsetType) -> bool + 'a,
-) -> Box<dyn Iterator<Item = PointOffsetType> + 'a> {
+) -> Box<dyn Iterator<Item = PointOffsetType> + 'a>
+where
+    H::Value: Clone,
+{
     let smallest_posting_idx = postings
         .iter()
         .enumerate()
         .min_by_key(|(_idx, posting)| posting.len())
         .map(|(idx, _posting)| idx)
         .unwrap();
```

Comment:
Unrelated to this PR, but have we considered sorting all posting lists, so that the smallest posting lists are checked first?